pilot/pkg/networking/core/listener_builder.go

```diff
@@ -163,6 +163,10 @@ func (lb *ListenerBuilder) patchOneListener(l *listener.Listener, ctx networking
 }
 
 func (lb *ListenerBuilder) patchListeners() {
+	if lb.node.IsWaypointProxy() {
+		// EnvoyFilter is not implemented for waypoints
+		return
```

Comment:
nit: log here or above

Reply:
This code is unconditionally called; I don't think the presence of a envoyfilter and a waypoint in the mesh should give them log spam. Also, we cannot even do "warn if there is an attached one" since most users have global envoyfilters